k8up/templates/deployment.yaml

```diff
@@ -1,36 +1,63 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ template "k8up-operator.fullname" $ }}
+  name: {{ include "k8up.fullname" . }}
   labels:
-    app: {{ template "k8up-operator.name" $ }}
-    chart: {{ template "k8up-operator.chart" $ }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+{{ include "k8up.labels" . | indent 4 }}
 spec:
+  # Running more than 1 replica is not supported, thus not configurable.
   replicas: 1
+  strategy:
+    type: Recreate
   selector:
     matchLabels:
-      app: {{ template "k8up-operator.name" $ }}
+      {{- include "k8up.selectorLabels" . | nindent 6 }}
   template:
     metadata:
       labels:
-        app: {{ template "k8up-operator.name" $ }}
+        {{- include "k8up.selectorLabels" . | nindent 8 }}
     spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
       containers:
         - name: k8up-operator
-          image: {{ .Values.k8up_operator.image }}
-          imagePullPolicy: Always
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
           env:
-{{- if .Values.k8up_operator.envVars }}
-{{ toYaml .Values.k8up_operator.envVars | indent 10 }}
-{{- end }}
-{{- if .Values.k8up_operator.extraEnvVars}}
-{{ toYaml .Values.k8up_operator.extraEnvVars | indent 10 }}
-{{- end }}
+          {{- with .Values.k8up.backupImage }}
+            - name: BACKUP_IMAGE
+              value: "{{ .repository}}:{{ .tag }}"
+          {{- end }}
+          {{- if .Values.k8up.envVars }}
+            {{- toYaml .Values.k8up.envVars | nindent 12 }}
+          {{- end }}
           ports:
-          - containerPort: 8080
-            protocol: TCP
+            - name: http
+              containerPort: 8080
+          livenessProbe:
+            httpGet:
+              path: /metrics
+              port: http
+            initialDelaySeconds: 2
```

Comment:
I think that's a little bit too optimistic. Let's put at least 30s so if the operator takes a while starting up it's not ending up in a kill loop.

Reply:
I have no idea how long the startup takes. I was under impression that the prometheus endpoint should be up fairly quickly. Yes there might be stuff in the background that still needs warmup. @Kidswiss how long does a startup usually take? Are 30s justified?

Reply:
True but regardless, I don't hink a delay shorter than 30s ever makes sense. Think about a cluster which is under some resource pressure and it just takes a while to start serving traffic. In such a situation it makes matters worse if the pod keeps getting killed...